### PR TITLE
Feature/claim button

### PR DESCRIPTION
**Pull request by memacius32**
- Adding Claim Button for user participating Farms.
- Adding wrapper for PoolCardTri for only fetching staking info on farms in which user is staking.

Button icon and design is of course open to suggestions

https://user-images.githubusercontent.com/96993065/150833627-516352fc-167d-4e1d-b7ad-fa86c65becde.mov

### CODE REVIEW

**Comment by rej156** [comment]
lgtm

prefer display flex and flex-direction column so it's two rows and change the button icon to the word "Claim" just to keep it consistent re: CTAs

either way it's fine if this gets deployed, works end of the day

**Comment by 0xchain1** [comment]
agree, if the button can say claim that would be great

**Comment by specialist0x** [comment]
Would we be able to get away with something like this:

image

**Comment by specialist0x** [comment]
> lgtm
> 
> prefer display flex and flex-direction column so it's two rows and change the button icon to the word "Claim" just to keep it consistent re: CTAs
> 
> either way it's fine if this gets deployed, works end of the day

Would that make staked cards have a larger vertical height than unstaked cards?

**Comment by rej156** [comment]
> > lgtm
> > prefer display flex and flex-direction column so it's two rows and change the button icon to the word "Claim" just to keep it consistent re: CTAs
> > either way it's fine if this gets deployed, works end of the day
> 
> Would that make staked cards have a larger vertical height than unstaked cards?

yeah it would 😬

**Comment by 0xchain1** [comment]
> > lgtm
> > prefer display flex and flex-direction column so it's two rows and change the button icon to the word "Claim" just to keep it consistent re: CTAs
> > either way it's fine if this gets deployed, works end of the day
> 
> Would that make staked cards have a larger vertical height than unstaked cards?

Another way to do it is
- Replace "Manage" with "Claim"
- Create an icon for manage. Can be gear box or something. On clicking the icon, takes user to the main manage page

**Comment by memacius32** [comment]
> lgtm
> 
> prefer display flex and flex-direction column so it's two rows and change the button icon to the word "Claim" just to keep it consistent re: CTAs
> 
> either way it's fine if this gets deployed, works end of the day

For consideration, that would look something like this:
 (We could also make the pool heading fixed on the top ofc)

Screen Shot 2022-01-25 at 16 04 32




Although I am liking this approach:
 


> > > lgtm
> > > prefer display flex and flex-direction column so it's two rows and change the button icon to the word "Claim" just to keep it consistent re: CTAs
> > > either way it's fine if this gets deployed, works end of the day
> > 
> > 
> > Would that make staked cards have a larger vertical height than unstaked cards?
> 
> Another way to do it is
> 
> * Replace "Manage" with "Claim"
> * Create an icon for manage. Can be gear box or something. On clicking the icon, takes user to the main manage page

**Comment by rej156** [comment]
aye defo go ahead with what 0xchain1 suggested

thanks for trying out my suggestion and sharing chad 😊

File: src/components/earn/PoolCardTri.tsx [hunk 1]
```diff
@@ -1,118 +1,82 @@
...
-  `}
-
-  ${({ theme }) => theme.mediaWidth.upToSmall`
-        padding: 4px;
-        border-radius: 5px;
-    `};
-`
-
-export default function PoolCardTRI({
+const ZERO = JSBI.BigInt(0)
```
Comment (specialist0x) [inline src/components/earn/PoolCardTri.tsx:46]: We should put this in some central location, it's used in hella places

File: src/components/earn/PoolCardTri.tsx [hunk 2]
```diff
@@ -128,15 +92,24 @@ export default function PoolCardTRI({
...
-            {isStaking ? t('earn.manage') : t('earn.deposit')}
-          </Button>
+          <StyledActionsContainer>
+            {enableClaimButton && (
+              <ButtonGold padding="8px" borderRadius="8px" onClick={enableModal} width="32px">
+                <ClaimIcon style={{ height: 30, width: 30 }} />
+              </ButtonGold>
+            )}
+
+            <Button
```
Comment (specialist0x) [inline src/components/earn/PoolCardTri.tsx:102]: nit
```suggestion
            )}
            <Button
```

File: src/components/earn/PoolCardTri.tsx [hunk 3]
```diff
@@ -159,3 +132,34 @@ export default function PoolCardTRI({
...
   )
 }
+
+const StakingPoolCardTRI = (props: PoolCardTriProps) => {
+  const { version } = props
+
+  const stakingInfo = useSingleFarm(Number(version))
+  const { earnedAmount } = stakingInfo
+
+  const amountIsClaimable = !earnedAmount == null || !earnedAmount?.equalTo(ZERO)
```
Comment (specialist0x) [inline src/components/earn/PoolCardTri.tsx:142]:
```suggestion
  const amountIsClaimable = (earnedAmount?.ge(ZERO) ?? false) === true
```

**Comment by specialist0x** [comment]
> > > lgtm
> > > prefer display flex and flex-direction column so it's two rows and change the button icon to the word "Claim" just to keep it consistent re: CTAs
> > > either way it's fine if this gets deployed, works end of the day
> > 
> > 
> > Would that make staked cards have a larger vertical height than unstaked cards?
> 
> Another way to do it is
> 
> * Replace "Manage" with "Claim"
> * Create an icon for manage. Can be gear box or something. On clicking the icon, takes user to the main manage page

+1

Gear (without claim) is also easily understood

**Comment by rej156** [comment]
🔥 🚒

**Comment by specialist0x** [comment]
Updated to use icon for Manage, and show claim button:

image

Claim button takes a sec to load, but otherwise LGTM

**Review by 0xchain1 — APPROVED** [review]
looks great

**Pull request merged**